src/database/nodeinfostorage.cpp

```diff
@@ -0,0 +1,209 @@
+        now;
+}
+
+Poco::JSON::Object NodeInfoStorage::ConvertNodeInfoToJSON(const aos::NodeInfo& nodeInfo)
+{
+    Poco::JSON::Object object;
+
+    object.set("type", nodeInfo.mType.CStr());
+    object.set("name", nodeInfo.mName.CStr());
+    object.set("osType", nodeInfo.mOSType.CStr());
+    object.set("cpuInfo", ConvertCpuInfoToJSON(nodeInfo.mCPUs));
+    object.set("partitions", ConvertPartitionInfoToJSON(nodeInfo.mPartitions));
+    object.set("attrs", ConvertAttributesToJSON(nodeInfo.mAttrs));
+    object.set("maxDMIPS", nodeInfo.mMaxDMIPS);
+    object.set("totalRAM", nodeInfo.mTotalRAM);
+
+    return object;
+}
+
+void NodeInfoStorage::ConvertNodeInfoFromJSON(const Poco::JSON::Object& object, aos::NodeInfo& dst)
+{
+    dst.mType   = object.getValue<std::string>("type").c_str();
+    dst.mName   = object.getValue<std::string>("name").c_str();
+    dst.mOSType = object.getValue<std::string>("osType").c_str();
+
+    const auto& cpuInfo = object.get("cpuInfo").extract<Poco::JSON::Array::Ptr>();
+    AssertNotNull(cpuInfo);
+    ConvertCpuInfoFromJSON(*cpuInfo, dst.mCPUs);
+
+    const auto& partitions = object.get("partitions").extract<Poco::JSON::Array::Ptr>();
+    AssertNotNull(partitions);
+    ConvertPartitionInfoFromJSON(*partitions, dst.mPartitions);
+
+    const auto& attributes = object.get("attrs").extract<Poco::JSON::Array::Ptr>();
+    AssertNotNull(attributes);
+    ConvertAttributesFromJSON(*attributes, dst.mAttrs);
+
+    dst.mMaxDMIPS = object.getValue<float>("maxDMIPS");
+    dst.mTotalRAM = object.getValue<size_t>("totalRAM");
+}
+
+Poco::JSON::Array NodeInfoStorage::ConvertCpuInfoToJSON(const aos::Array<aos::CPUInfo>& cpuInfo)
+{
+    Poco::JSON::Array dst;
+
+    for (const auto& srcItem : cpuInfo) {
+        Poco::JSON::Object pocoItem;
+
+        pocoItem.set("id", srcItem.mID);
+        pocoItem.set("modelName", srcItem.mModelName.CStr());
+        pocoItem.set("numCores", srcItem.mNumCores);
+        pocoItem.set("numThreads", srcItem.mNumThreads);
+        pocoItem.set("arch", srcItem.mArch.CStr());
+        pocoItem.set("archFamily", srcItem.mArchFamily.CStr());
+
+        dst.add(pocoItem);
+    }
+
+    return dst;
+}
+
+void NodeInfoStorage::ConvertCpuInfoFromJSON(const Poco::JSON::Array& src, aos::Array<aos::CPUInfo>& dst)
+{
+    for (const auto& srcItem : src) {
+        const auto& cpuInfo = srcItem.extract<Poco::JSON::Object::Ptr>();
+        AssertNotNull(cpuInfo);
+
+        aos::CPUInfo dstItem;
+
+        dstItem.mID         = cpuInfo->getValue<size_t>("id");
+        dstItem.mModelName  = cpuInfo->getValue<std::string>("modelName").c_str();
+        dstItem.mNumCores   = cpuInfo->getValue<size_t>("numCores");
+        dstItem.mNumThreads = cpuInfo->getValue<size_t>("numThreads");
+        dstItem.mArch       = cpuInfo->getValue<std::string>("arch").c_str();
+        dstItem.mArchFamily = cpuInfo->getValue<std::string>("archFamily").c_str();
+
+        dst.PushBack(dstItem);
```

Comment:
Consider adding check for PushBack result

Reply:
done